Riot/Modules/Analytics/AnalyticsSpaceTracker.swift

```diff
@@ -0,0 +1,72 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+class AnalyticsSpaceTracker {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let lastNumberOfSpaces: String = "AnalyticsSpaceTracker.lastNumberOfSpaces"
+    }
+    
+    // Last number of spaces tracked
+    private var lastNumberOfSpaces: Int? {
+        get {
+            guard let value = UserDefaults.standard.value(forKey: Constants.lastNumberOfSpaces) as? NSNumber else {
```

Comment:
Probably not important for now as I think the space service only runs in the main target, but it might potentially be useful to store this in `RiotSettings` or `RiotSettings.defaults` so that it is available in another target if ever needed.

For the FTUEUseCaseSelection I added a `UserSessionProperties` class that we can add user properties to on a per user basis (pre-planning for multi-account at some point) and it will store them together. Might be possible for you to use that if you wanted.

Reply:
Good point. My first thought was to not making `RiotSettings` too much but I forgot about targets. I'll make the update.